plugins/trace_api_plugin/include/eosio/trace_api/request_handler.hpp

```diff
@@ -6,12 +6,31 @@
 #include <eosio/trace_api/common.hpp>
 
 namespace eosio::trace_api {
-   using data_handler_function = std::function<fc::variant(const action_trace_v0&, const yield_function&)>;
+   using data_handler_function_v0 = std::function<fc::variant(const action_trace_v0&, const yield_function&)>;
+   using data_handler_function_v1 = std::function<fc::variant(const action_trace_v1&, const yield_function&)>;
+   using return_handler_function = std::function<fc::variant(const action_trace_v1&, const yield_function&)>;
+   using transaction_trace_variant = std::variant<transaction_trace_v0, transaction_trace_v1, transaction_trace_v2, transaction_trace_v3>;
+   using action_trace_variant = std::variant<action_trace_v0, action_trace_v1>;
+   using vec_transaction_trace_variant = std::vector<transaction_trace_variant>;
+   using vec_action_trace_variant = std::vector<action_trace_variant>;
 
+   enum trace_api_type_enum {
+   	e_block_trace_v0,
+	e_block_trace_v1,
+	e_block_trace_v2,
+	e_block_trace_v3,
+	e_transaction_trace_v0,
+	e_transaction_trace_v1,
+	e_transaction_trace_v2,
+	e_transaction_trace_v3,
+	e_action_trace_v0,
+	e_action_trace_v1
+   };
+   
    namespace detail {
       class response_formatter {
       public:
-         static fc::variant process_block( const data_log_entry& trace, bool irreversible, const data_handler_function& data_handler, const yield_function& yield );
+         static fc::variant process_block( const data_log_entry& trace, bool irreversible, const data_handler_function_v0& data_handler_v0, const data_handler_function_v1 & data_handler_v1,   const return_handler_function & return_handler,  const yield_function& yield );
```

Comment:
Yes, now as above.   only different is  fc::optional instead of  std::optional for compile issue.   
currently is: std::function<std::tuple<fc::variant, fc::optional<fc::variant>>(const action_trace_t&, const yield_function&)>